query/cursor.go

```diff
@@ -198,6 +200,9 @@ func (cur *scannerCursorBase) Scan(row *Row) bool {
 		row.Values = make([]interface{}, len(cur.columns))
 	}
 
+	// Set the timestamp in the map so date_part can access it
+	cur.m["time"] = row.Time
+
```

Comment:
Do we add this to the map only when `DATEPART` is being used? Or is it added even when there is no `DATEPART`?